kyuubi-server/src/main/scala/org/apache/kyuubi/credentials/HadoopCredentialsManager.scala

```diff
@@ -239,7 +246,9 @@ class HadoopCredentialsManager private (name: String) extends AbstractService(na
         try {
           promise.trySuccess(updateCredentials(userRef))
 
-          scheduleRenewal(userRef, renewalInterval)
+          if (userCredentialsRefMap.containsKey(userRef.getAppUser)) {
```

Comment:
Sorry for late reply. But considering the following scenario, I think we should also check whether `userRef` in `userCredentialsRefMap` has been replaced by a new one .

1. `userRef` was waiting for next renewal. 
2. `userRef` expired and was removed from `userCredentialsRefMap`
3.  A new `userRef` was created because of user visiting.
4. Renewal of old `userRef` started again.

If we do not check, both old and new `userRef` will renew periodically.

Reply:
@zhouyifan279
Thanks. I will do a follow-up commit.